src/compiler/core.ts

```diff
@@ -174,23 +262,47 @@ namespace ts {
             clear(): void {
                 this.data = createDictionaryObject<T>();

```

Comment:
The type `T` here is wrong. The compiler didn't catch that because `MapLike<T>` is a very weak type.

Reply:
Thank you, fixed.